src/pages/workspaces/workspace/notebooks/modals/CloudEnvironmentModal.js

```diff
@@ -0,0 +1,404 @@
+          onClick: () => {
+            ((toolLabel === tools.Jupyter.label || toolLabel === tools.RStudio.label) && currentRuntime?.status === 'Stopped' ? () => startApp(toolLabel) : () => {})()
+            onDismiss()
+          }
+        }
+      }]
+    )
+  }
+
+
+  const renderToolButtons = toolLabel => {
+    const doesCloudEnvForToolExist = currentRuntimeTool === toolLabel || (currentApp && toolLabel === tools.galaxy.label)
+    const isCloudEnvForToolDisabled = isCloudEnvModalDisabled(toolLabel)
+    return h(Fragment, [
+      div({ style: toolPanelStyles }, [
+        // Label at the top for each tool
+        div({ style: toolLabelStyles }, [
+          img({
+            src: getToolIcon(toolLabel),
+            style: { height: 20 }
+          }),
+          getCostForTool(toolLabel)
+        ]),
+        // Cloud environment button
+        div({ style: toolButtonDivStyles }, [
+          h(Clickable, {
+            'aria-label': `${toolLabel} Env`,
+            style: {
+              ...toolButtonStyles,
+              color: !isCloudEnvForToolDisabled ? colors.accent() : colors.dark(0.7)
+            },
+            hover: { backgroundColor: colors.accent(0.2) },
+            tooltip: Utils.cond([isCloudEnvForToolDisabled, () => 'Edit disabled, processing'],
+              [doesCloudEnvForToolExist, () => 'Edit existing Environment'],
+              [!doesCloudEnvForToolExist, () => 'Create new Environment (may overwrite existing)']),
+            disabled: isCloudEnvForToolDisabled,
+            tooltipDelay: 100,
+            onClick: () => setViewMode(toolLabel)
+          }, [
+            img({ src: cloudIcon, style: { height: 20, width: 20, opacity: isCloudEnvForToolDisabled ? 0.4 : 1 } }), //TODO: why doesn't this icon disable properly color-wise?
+            span('Cloud'),
+            span('Environment')
+          ]),
+          // Status button with stop/start functionality
+          renderStatusClickable(toolLabel),
+          // Launch
+          h(Clickable, { ...getToolLaunchClickableProps(toolLabel) }, [
+            icon('rocket', { size: 20 }),
+            span('Launch'),
+            span(_.capitalize(toolLabel))
+          ])
+        ])
+      ])
+    ])
+  }
+
+  const NEW_JUPYTER_MODE = tools.Jupyter.label
+  const NEW_RSTUDIO_MODE = tools.RStudio.label
+  const NEW_GALAXY_MODE = tools.galaxy.label
+
+  const getView = () => Utils.switchCase(viewMode,
+    [NEW_JUPYTER_MODE, () => renderNewRuntimeModal(NEW_JUPYTER_MODE)],
+    [NEW_RSTUDIO_MODE, () => renderNewRuntimeModal(NEW_RSTUDIO_MODE)],
+    [NEW_GALAXY_MODE, renderNewGalaxyModal],
+    [Utils.DEFAULT, renderDefaultPage]
+  )
+
+  const width = Utils.switchCase(viewMode,
+    [NEW_JUPYTER_MODE, () => 675],
+    [NEW_RSTUDIO_MODE, () => 675],
+    [NEW_GALAXY_MODE, () => 675],
+    [Utils.DEFAULT, () => 430]
+  )
+
+  const modalBody = h(Fragment, [
+    h(TitleBar, {
+      id: titleId,
+      title: 'Cloud environment details',
```

Comment:
Would 
```suggestion
      title: 'Cloud Environment Details',
```
be more consistent since we use title case for `Cloud Environment`

![image](https://user-images.githubusercontent.com/5438223/127922714-cf9d1c21-acf8-4d01-9dd7-abfcc3f3834d.png)